spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DefaultEnvironmentPostProcessor.java

```diff
@@ -58,7 +64,17 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 
 		contributeDefaults(internalDefaults, serverDefaultsResource);
 		contributeDefaults(defaults, serverResource);
-
+		String hostName = null;
+		try {
+			hostName = InetAddress.getLocalHost().getHostAddress();
+		}
+		catch (UnknownHostException e) {
+			// ignore
+		}
+		String dataflowServerPort = environment.getProperty("server.port");
+		if (StringUtils.hasText(hostName) && StringUtils.hasText(dataflowServerPort)) {
+			defaults.put(DataFlowPropertyKeys.PREFIX + "server.uri", String.format("http://%s:%s", hostName, dataflowServerPort));
```

Comment:
I'm not sure this would be useful in a distributed environment - we need to provide the routable host/ip-address and most likely the server doesn't have access to this without some explicit configuration. I think we should check for a `spring.cloud.dataflow.server.uri` property and if it is available use it, if it's not set just rely on the CTR to use its default.